packages/app-store/applecalendar/api/add.ts

```diff
@@ -18,9 +18,22 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
       select: {
         email: true,
         id: true,
+        credentials: {
+          where: {
+            type: "apple_calendar",
+          },
+        },
       },
     });
 
+    const alreadyExists = user.credentials.find((credential) => {
+      const decryptedCredential = JSON.parse(symmetricDecrypt(credential.key?.toString() || "", process.env.CALENDSO_ENCRYPTION_KEY || ""));
+      return decryptedCredential.username === username;
+    })
+
+
+    if (alreadyExists) return res.status(409).json({ message: "account_already_linked" });
```

Comment:
I would try to do something like 

`if(alreadyExists) && alreadyExists.password === password { return a 409 }`

`if(alreadyExists) && alreadyExists.password !== password { `
` users app specific password expired and so we update it in the credentials table`
`}`

Reply:
Implemented